### PR TITLE
chore(ingest): update doc & log detail

### PR DESCRIPTION
**Pull request by HuanjieGuo**
This is the first time I run the ingestion module.

* DOC has a little bit mismatching with the code.
* NPE check
* add the full path message in the exception

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/configuration/config_loader.py [hunk 1]
```diff
@@ -149,7 +149,8 @@ def load_config_file(
                 ) from e
         else:
             if not config_file_path.is_file():
-                raise ConfigurationError(f"Cannot open config file {config_file_path}")
+                abs_path = os.path.abspath(config_file_path)
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/configuration/config_loader.py:152]: can use .resolve() here

File: metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py [hunk 2]
```diff
@@ -58,6 +58,7 @@ def compute_stats(self) -> None:
 
 
 def _get_urn(record_envelope: RecordEnvelope) -> Optional[str]:
+    assert record_envelope is not None
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py:61]: the type annotations say that none will never be passed in here, so it feels like the issue is the caller

do you have a stack trace of where this was called with a null argument?

File: metadata-ingestion/src/datahub/utilities/advanced_thread_executor.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def __init__(self, max_workers: int, max_pending: int) -> None:
 
         It works similarly to a ThreadPoolExecutor, with the following changes:
         - At most one request per partition key will be executing at a time.
-        - If the number of pending requests exceeds the threshold, the submit call
+        - If the number of pending requests exceeds the threshold, the submitted call
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/utilities/advanced_thread_executor.py:35]: maybe this is more clear
```suggestion
        - If the number of pending requests exceeds the threshold, the submit() call
```

File: metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py (hunk 2 above)
Comment (HuanjieGuo) [reply metadata-ingestion/src/datahub/ingestion/sink/datahub_rest.py]: Do not have it. Just read the code and found that it directly use the field in a instance. Do not know that python type annotation will hint that it cannot be None.

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**